packages/compiler/src/myst.ts

```diff
@@ -36,5 +39,48 @@ export function parsePanel(contents: string): Array<MystCard> {
     parsedCard.body = body;
     splitCards.push(parsedCard);
   }
-  return splitCards;
+
+  let myPanel: MystPanel = {cards: splitCards}
+  if (Object.keys(panelStyling).length !== 0) {
+    myPanel = {...myPanel, style: panelStyling}
+  }
+  return myPanel
+}
+
+export function parsePanelStyling(contents: string) {
+  const contentLines = contents.split("\n")
+  // TO ASK: This regex was a PITA. 
+  // Is there a way to get it to do this without having to split contents by newline?
+  const panelStylingRegex = /:[a-z]+:\s[a-z\s\-0-9]+/i;
```

Comment:
Not sure if it's helpful but you can see what the python implementation of this parser here: https://github.com/executablebooks/MyST-Parser/blob/master/myst_parser/parse_directives.py#L121

Looks like they just consume lines starting with `:` until there aren't any more of them.

Reply:
I’m doing something similar now but since I don’t have a yaml parser, I still have a function to deal with parsing the block once it’s extracted.